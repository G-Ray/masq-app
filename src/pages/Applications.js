import React from 'react';

import Card from '../components/Card/Card';

import './Applications.css';

export default function Applications(props) {
  const apps = [
    {
      name: 'Qwant Maps',
      color: '#01cbd9',
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.",
      enabled: true
    },
    {
      name: 'Qwant Shopping',
      color: '#a3005c',
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.",
      enabled: false,
      image: 'https://images.pexels.com/photos/592753/pexels-photo-592753.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb'
    }
  ];

  const connectedApps = [
    {
      name: 'Qwant Music',
      color: '#a3005c',
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.",
      enabled: true,
      image: 'https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb'
    }
  ];

  return (
    <div style={{ backgroundColor: '#f5f7fa', paddingTop: '1px' }}>

      <h1 style={{marginLeft: '16px'}}>Currently active Applications</h1>
      <div className="Applications">
        {apps.map((app, index) => (
          <Card key={index} title={app.name} description={app.description} color={app.color} enabled={app.enabled} image={app.image} >
          </Card>
        ))}
      </div>

      <h1 style={{marginLeft: '16px'}}>Currently connected to your Masq</h1>
      <div className="Applications">
        {connectedApps.map((app, index) => (
          <Card key={index} title={app.name} description={app.description} color={app.color} enabled={app.enabled} image={app.image} >
          </Card>
        ))}
      </div>
    </div>
  );
}
