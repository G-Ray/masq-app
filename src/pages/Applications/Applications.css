.Applications {
   display: flex;
   flex-direction: column;
   align-items: center;
   justify-content: space-evenly;
}

.Applications .permissions {
  color: #191919;
  font-size: 10px;
  font-weight: 600;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.Applications .more {
  display: flex;
  align-items: center;
}

.Applications .caret-icon {
  vertical-align: middle;
  display: table-cell;
  padding-left: 8px;
  height: 5px;
}
