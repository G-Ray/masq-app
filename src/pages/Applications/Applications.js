import React from 'react'

import { Chevron } from 'icons'
import { Card, Separator } from 'components'

import './Applications.css'

function AppRow (props) {
  const { app } = props
  const style = app.enabled ? { color: app.color } : {}
  return (
    <Card title={app.name} description={app.description} color={app.color} enabled={app.enabled} image={app.image} onChecked={props.onChecked}>
      <div className='permissions'>
        <p>5 PERMISSIONS</p>
        <div className='more'>
          <p style={style}>SEE MORE</p>
          <Chevron className='caret-icon' style={style} />
        </div>
      </div>
    </Card>
  )
}

function AppRows (props) {
  const { applications, active } = props
  return (
    applications.map((app, index) => (
      app.active === active
        ? (
          <div key={index} >
            <AppRow app={app} onChecked={() => props.onChecked(index)} />
            <Separator />
          </div>
        )
        : null
    ))
  )
}

export default function Applications (props) {
  const { applications } = props

  return (
    <div>
      <h2 style={{marginLeft: '16px'}}>Currently active Applications</h2>
      <div className='Applications'>
        <AppRows applications={applications} active onChecked={props.onChecked} />
      </div>

      <h2 style={{marginLeft: '16px'}}>Currently connected to your Masq</h2>
      <div className='Applications'>
        <AppRows applications={applications} onChecked={props.onChecked} />
      </div>
    </div>
  )
}
