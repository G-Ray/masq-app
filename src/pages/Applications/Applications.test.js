import React from 'react'
import renderer from 'react-test-renderer'
import Applications from './Applications'

it('renders Applications page with one active app', () => {
  const applications = [
    {
      name: 'Qwant Music',
      color: '#5c00f3',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.',
      enabled: true,
      image: 'https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb',
      active: true
    }
  ]
  const component = renderer.create(<Applications applications={applications} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Applications page with one inactive app', () => {
  const applications = [
    {
      name: 'Qwant Music',
      color: '#5c00f3',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.',
      enabled: true,
      image: 'https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb'
    }
  ]
  const component = renderer.create(<Applications applications={applications} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
