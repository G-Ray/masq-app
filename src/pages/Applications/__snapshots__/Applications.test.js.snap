// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Applications page with one active app 1`] = `
<div>
  <h2
    style={
      Object {
        "marginLeft": "16px",
      }
    }
  >
    Currently active Applications
  </h2>
  <div
    className="Applications"
  >
    <div>
      <div
        className="Card"
      >
        <div
          className="image"
          style={
            Object {
              "backgroundImage": "url(https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb)",
            }
          }
        />
        <div
          className="flex"
        >
          <div
            className="content"
          >
            <div
              className="CardHeader"
            >
              <div
                className="line"
                style={
                  Object {
                    "backgroundColor": "#5c00f3",
                  }
                }
              />
              <div
                className="SwitchButton"
              >
                <label
                  style={
                    Object {
                      "color": "#5c00f3",
                    }
                  }
                >
                  ENABLED
                </label>
                <div
                  className=""
                  id="switch-container"
                  onClick={[Function]}
                >
                  <div
                    className=""
                    id="background"
                    style={
                      Object {
                        "backgroundColor": "#5c00f3",
                      }
                    }
                  />
                  <div
                    className=""
                    id="cursor"
                    style={
                      Object {
                        "transform": "translate3d(26px, 0, 0)",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <h1>
              Qwant Music
            </h1>
            <p
              className="description"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.
            </p>
            <div
              className="permissions"
            >
              <p>
                5 PERMISSIONS
              </p>
              <div
                className="more"
              >
                <p
                  style={
                    Object {
                      "color": "#5c00f3",
                    }
                  }
                >
                  SEE MORE
                </p>
                <svg
                  className="caret-icon"
                  height={10}
                  style={
                    Object {
                      "color": "#5c00f3",
                    }
                  }
                  viewBox="0 0 14 10"
                  width={14}
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <defs>
                    <path
                      d="M13.7 1.7l-6 6c-.2.2-.4.3-.7.3-.3 0-.5-.1-.7-.3l-6-6C-.1 1.3-.1.7.3.3c.4-.4 1-.4 1.4 0L7 5.6 12.3.3c.4-.4 1-.4 1.4 0 .4.4.4 1 0 1.4z"
                      id="chevron-svg"
                    />
                  </defs>
                  <use
                    fill="currentColor"
                    fillRule="nonzero"
                    xlinkHref="#chevron-svg"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        id="separator"
        style={
          Object {
            "marginTop": "16px",
          }
        }
      />
    </div>
  </div>
  <h2
    style={
      Object {
        "marginLeft": "16px",
      }
    }
  >
    Currently connected to your Masq
  </h2>
  <div
    className="Applications"
  />
</div>
`;

exports[`renders Applications page with one inactive app 1`] = `
<div>
  <h2
    style={
      Object {
        "marginLeft": "16px",
      }
    }
  >
    Currently active Applications
  </h2>
  <div
    className="Applications"
  />
  <h2
    style={
      Object {
        "marginLeft": "16px",
      }
    }
  >
    Currently connected to your Masq
  </h2>
  <div
    className="Applications"
  >
    <div>
      <div
        className="Card"
      >
        <div
          className="image"
          style={
            Object {
              "backgroundImage": "url(https://images.pexels.com/photos/63703/pexels-photo-63703.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb)",
            }
          }
        />
        <div
          className="flex"
        >
          <div
            className="content"
          >
            <div
              className="CardHeader"
            >
              <div
                className="line"
                style={
                  Object {
                    "backgroundColor": "#5c00f3",
                  }
                }
              />
              <div
                className="SwitchButton"
              >
                <label
                  style={
                    Object {
                      "color": "#5c00f3",
                    }
                  }
                >
                  ENABLED
                </label>
                <div
                  className=""
                  id="switch-container"
                  onClick={[Function]}
                >
                  <div
                    className=""
                    id="background"
                    style={
                      Object {
                        "backgroundColor": "#5c00f3",
                      }
                    }
                  />
                  <div
                    className=""
                    id="cursor"
                    style={
                      Object {
                        "transform": "translate3d(26px, 0, 0)",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <h1>
              Qwant Music
            </h1>
            <p
              className="description"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.
            </p>
            <div
              className="permissions"
            >
              <p>
                5 PERMISSIONS
              </p>
              <div
                className="more"
              >
                <p
                  style={
                    Object {
                      "color": "#5c00f3",
                    }
                  }
                >
                  SEE MORE
                </p>
                <svg
                  className="caret-icon"
                  height={10}
                  style={
                    Object {
                      "color": "#5c00f3",
                    }
                  }
                  viewBox="0 0 14 10"
                  width={14}
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <defs>
                    <path
                      d="M13.7 1.7l-6 6c-.2.2-.4.3-.7.3-.3 0-.5-.1-.7-.3l-6-6C-.1 1.3-.1.7.3.3c.4-.4 1-.4 1.4 0L7 5.6 12.3.3c.4-.4 1-.4 1.4 0 .4.4.4 1 0 1.4z"
                      id="chevron-svg"
                    />
                  </defs>
                  <use
                    fill="currentColor"
                    fillRule="nonzero"
                    xlinkHref="#chevron-svg"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        id="separator"
        style={
          Object {
            "marginTop": "16px",
          }
        }
      />
    </div>
  </div>
</div>
`;
